src/Components/NewTodo.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { Fragment, useState } from "react";
+import { useDispatch } from "react-redux";
+import { addTodo } from "services/actions/addTodo";
+
+export default function NewTodo(props) {
+  const { onNewSave } = props;
+  const [todoText, setTodoText] = useState(null);
+
+  const dispatch = useDispatch();
```

Comment:
this types of functions required to be top of function to to avoid unexpected re-rendering issue.

Reply:
addressed